{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "5eae01bd",
   "metadata": {},
   "source": [
    "## Reading cleaned VCF"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "414f9815",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[W::vcf_parse] Contig '1' is not defined in the header. (Quick workaround: index the file with tabix.)\n",
      "[W::vcf_parse] Contig '2' is not defined in the header. (Quick workaround: index the file with tabix.)\n",
      "[W::vcf_parse] Contig '3' is not defined in the header. (Quick workaround: index the file with tabix.)\n",
      "[W::vcf_parse] Contig '4' is not defined in the header. (Quick workaround: index the file with tabix.)\n",
      "[W::vcf_parse] Contig '5' is not defined in the header. (Quick workaround: index the file with tabix.)\n",
      "[W::vcf_parse] Contig '6' is not defined in the header. (Quick workaround: index the file with tabix.)\n",
      "[W::vcf_parse] Contig '7' is not defined in the header. (Quick workaround: index the file with tabix.)\n",
      "[W::vcf_parse] Contig '8' is not defined in the header. (Quick workaround: index the file with tabix.)\n",
      "[W::vcf_parse] Contig '9' is not defined in the header. (Quick workaround: index the file with tabix.)\n",
      "[W::vcf_parse] Contig '10' is not defined in the header. (Quick workaround: index the file with tabix.)\n",
      "[W::vcf_parse] Contig '11' is not defined in the header. (Quick workaround: index the file with tabix.)\n",
      "[W::vcf_parse] Contig '12' is not defined in the header. (Quick workaround: index the file with tabix.)\n",
      "[W::vcf_parse] Contig '13' is not defined in the header. (Quick workaround: index the file with tabix.)\n",
      "[W::vcf_parse] Contig '14' is not defined in the header. (Quick workaround: index the file with tabix.)\n",
      "[W::vcf_parse] Contig '15' is not defined in the header. (Quick workaround: index the file with tabix.)\n",
      "[W::vcf_parse] Contig '16' is not defined in the header. (Quick workaround: index the file with tabix.)\n",
      "[W::vcf_parse] Contig '17' is not defined in the header. (Quick workaround: index the file with tabix.)\n",
      "[W::vcf_parse] Contig '18' is not defined in the header. (Quick workaround: index the file with tabix.)\n",
      "[W::vcf_parse] Contig '19' is not defined in the header. (Quick workaround: index the file with tabix.)\n",
      "[W::vcf_parse] Contig '20' is not defined in the header. (Quick workaround: index the file with tabix.)\n",
      "[W::vcf_parse] Contig '21' is not defined in the header. (Quick workaround: index the file with tabix.)\n",
      "[W::vcf_parse] Contig '22' is not defined in the header. (Quick workaround: index the file with tabix.)\n",
      "[W::vcf_parse] Contig 'X' is not defined in the header. (Quick workaround: index the file with tabix.)\n",
      "[W::vcf_parse] Contig 'Y' is not defined in the header. (Quick workaround: index the file with tabix.)\n",
      "[W::vcf_parse] Contig 'MT' is not defined in the header. (Quick workaround: index the file with tabix.)\n",
      "[W::vcf_parse] Contig 'NT_113889.1' is not defined in the header. (Quick workaround: index the file with tabix.)\n",
      "[W::vcf_parse] Contig 'NT_187633.1' is not defined in the header. (Quick workaround: index the file with tabix.)\n",
      "[W::vcf_parse] Contig 'NT_187661.1' is not defined in the header. (Quick workaround: index the file with tabix.)\n",
      "[W::vcf_parse] Contig 'NT_187693.1' is not defined in the header. (Quick workaround: index the file with tabix.)\n",
      "[W::vcf_parse] Contig 'NW_009646201.1' is not defined in the header. (Quick workaround: index the file with tabix.)\n"
     ]
    }
   ],
   "source": [
    "from cyvcf2 import VCF\n",
    "import pandas as pd \n",
    "\n",
    "vcf = VCF('data/cleaned.vcf')\n",
    "\n",
    "rows = []\n",
    "for record in vcf:\n",
    "    info = record.INFO\n",
    "    rows.append({\n",
    "        'Chromosome': record.CHROM,\n",
    "        'Position': record.POS,\n",
    "        'Reference': record.REF,\n",
    "        'Alternate_Allele': \",\".join(map(str, record.ALT)),\n",
    "        \"Gene_Symbol\": info.get(\"GENEINFO\", \"\").split(\":\")[0] if info.get(\"GENEINFO\") else None,\n",
    "        \"Clinical_Significance\": info.get(\"CLNSIG\"),\n",
    "        \"Clinical_Review_Status\": info.get(\"CLNREVSTAT\"),\n",
    "        \"Clinical_Disease_Name\": info.get(\"CLNDN\"),\n",
    "        \"Missense_Variant\": info.get(\"MC\"),\n",
    "        \"Allele_Frequencies_ESP\": info.get(\"AF_ESP\"),\n",
    "        \"Allele_Frequencies_EXAC\": info.get(\"AF_EXAC\"),\n",
    "        \"Allele_Frequencies_TGP\": info.get(\"AF_TGP\")\n",
    "    })\n",
    "    \n",
    "df = pd.DataFrame(rows)\n",
    "\n",
    "# Save clean, readable TSV\n",
    "df.to_csv(\"data/genomics_clean.tsv\", sep=\"\\t\", index=False)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "041d0ba1",
   "metadata": {},
   "source": [
    "## Load the dataset"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "c9bff7bd",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/var/folders/jx/cc9brgl143z9wt4bc65wylz40000gn/T/ipykernel_91209/2591019169.py:1: DtypeWarning: Columns (0) have mixed types. Specify dtype option on import or set low_memory=False.\n",
      "  dataset = pd.read_csv('data/genomics_clean.tsv', sep ='\\t')\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Chromosome</th>\n",
       "      <th>Position</th>\n",
       "      <th>Reference</th>\n",
       "      <th>Alternate_Allele</th>\n",
       "      <th>Gene_Symbol</th>\n",
       "      <th>Clinical_Significance</th>\n",
       "      <th>Clinical_Review_Status</th>\n",
       "      <th>Clinical_Disease_Name</th>\n",
       "      <th>Missense_Variant</th>\n",
       "      <th>Allele_Frequencies_ESP</th>\n",
       "      <th>Allele_Frequencies_EXAC</th>\n",
       "      <th>Allele_Frequencies_TGP</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>66926</td>\n",
       "      <td>AG</td>\n",
       "      <td>A</td>\n",
       "      <td>OR4F5</td>\n",
       "      <td>Uncertain_significance</td>\n",
       "      <td>criteria_provided,_single_submitter</td>\n",
       "      <td>Retinitis_pigmentosa</td>\n",
       "      <td>SO:0001627|intron_variant</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1</td>\n",
       "      <td>69134</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>OR4F5</td>\n",
       "      <td>Likely_benign</td>\n",
       "      <td>criteria_provided,_single_submitter</td>\n",
       "      <td>not_specified</td>\n",
       "      <td>SO:0001583|missense_variant</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>1</td>\n",
       "      <td>69308</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>OR4F5</td>\n",
       "      <td>Uncertain_significance</td>\n",
       "      <td>criteria_provided,_single_submitter</td>\n",
       "      <td>not_specified</td>\n",
       "      <td>SO:0001583|missense_variant</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>1</td>\n",
       "      <td>69314</td>\n",
       "      <td>T</td>\n",
       "      <td>G</td>\n",
       "      <td>OR4F5</td>\n",
       "      <td>Uncertain_significance</td>\n",
       "      <td>criteria_provided,_single_submitter</td>\n",
       "      <td>not_specified</td>\n",
       "      <td>SO:0001583|missense_variant</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>1</td>\n",
       "      <td>69404</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>OR4F5</td>\n",
       "      <td>Uncertain_significance</td>\n",
       "      <td>criteria_provided,_single_submitter</td>\n",
       "      <td>not_specified</td>\n",
       "      <td>SO:0001583|missense_variant</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  Chromosome  Position Reference Alternate_Allele Gene_Symbol  \\\n",
       "0          1     66926        AG                A       OR4F5   \n",
       "1          1     69134         A                G       OR4F5   \n",
       "2          1     69308         A                G       OR4F5   \n",
       "3          1     69314         T                G       OR4F5   \n",
       "4          1     69404         T                C       OR4F5   \n",
       "\n",
       "    Clinical_Significance               Clinical_Review_Status  \\\n",
       "0  Uncertain_significance  criteria_provided,_single_submitter   \n",
       "1           Likely_benign  criteria_provided,_single_submitter   \n",
       "2  Uncertain_significance  criteria_provided,_single_submitter   \n",
       "3  Uncertain_significance  criteria_provided,_single_submitter   \n",
       "4  Uncertain_significance  criteria_provided,_single_submitter   \n",
       "\n",
       "  Clinical_Disease_Name             Missense_Variant  Allele_Frequencies_ESP  \\\n",
       "0  Retinitis_pigmentosa    SO:0001627|intron_variant                     NaN   \n",
       "1         not_specified  SO:0001583|missense_variant                     NaN   \n",
       "2         not_specified  SO:0001583|missense_variant                     NaN   \n",
       "3         not_specified  SO:0001583|missense_variant                     NaN   \n",
       "4         not_specified  SO:0001583|missense_variant                     NaN   \n",
       "\n",
       "   Allele_Frequencies_EXAC  Allele_Frequencies_TGP  \n",
       "0                      NaN                     NaN  \n",
       "1                      NaN                     NaN  \n",
       "2                      NaN                     NaN  \n",
       "3                      NaN                     NaN  \n",
       "4                      NaN                     NaN  "
      ]
     },
     "execution_count": 9,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "dataset = pd.read_csv('data/genomics_clean.tsv', sep ='\\t')\n",
    "dataset.head()"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Genomics (conda)",
   "language": "python",
   "name": "genomics"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.10.19"
  },
  "toc": {
   "base_numbering": 1,
   "nav_menu": {},
   "number_sections": true,
   "sideBar": true,
   "skip_h1_title": false,
   "title_cell": "Table of Contents",
   "title_sidebar": "Contents",
   "toc_cell": false,
   "toc_position": {
    "height": "calc(100% - 180px)",
    "left": "10px",
    "top": "150px",
    "width": "384px"
   },
   "toc_section_display": true,
   "toc_window_display": true
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
